permission_handler_android/android/src/main/java/com/baseflow/permissionhandler/PackageInfoHostApiImpl.java

```diff
@@ -0,0 +1,55 @@
+package com.baseflow.permissionhandler;
+
+import android.content.pm.PackageInfo;
+
+import androidx.annotation.NonNull;
+
+import com.baseflow.instancemanager.InstanceManager;
+import com.baseflow.permissionhandler.PermissionHandlerPigeon.PackageInfoHostApi;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+import io.flutter.plugin.common.BinaryMessenger;
+
+/**
+ * Host API implementation for `PackageInfo`.
+ *
+ * <p>This class may handle instantiating and adding native object instances that are attached to a
+ * Dart instance or handle method calls on the associated native class or an instance of the class.
+ */
+public class PackageInfoHostApiImpl implements PackageInfoHostApi {
+    // To ease adding additional methods, this value is added prematurely.
+    @SuppressWarnings({"unused", "FieldCanBeLocal"})
+    private final BinaryMessenger binaryMessenger;
```

Comment:
Thinking a bit ahead, I don't see we would need the `binaryMessenger` anywhere else except for the constructor of the class. The `binaryMessenger` should only be used to pass into Flutter API implementation classes and these are always created in the constructor. Therefore I feel we can remove this field (and also in already existing code).